app/utils/code-editor/format/format-editor-doc.js

```diff
@@ -1,23 +1,228 @@
+        // because im setting the formatting boolean to FALSE
+        formatting = false;
+        console.log('Stopped formatting. Ran into an unknown format', text);
+        doc.push(text);
+      }
+
+      const textBeforeMatch = text.slice(0, indexOfClosesMatch).trim();
+      if (lineBeforeValidationSummation.test(textBeforeMatch)) {
+        const endOfBlankNode = getFirstMatchWithFormat(
+          textBeforeMatch,
+          lineBeforeValidationSummation
+        );
+        const restValue = textBeforeMatch.slice(
+          endOfBlankNode.index + endOfBlankNode[0].length,
+          undefined
+        );
+        doc.push('\t' + endOfBlankNode[0]);
+        doc.push(restValue.trim());
+        text = text.replace(textBeforeMatch, '').trim();
+      } else {
+        doc.push(textBeforeMatch);
+        text = text.slice(indexOfClosesMatch, undefined).trim();
+      }
+    }
+  }
+
+  addIndentationToLines(doc);
+
+  return doc;
+}
+
+function addIndentationToLines(doc) {
+  let inValidationBlankNodes = false;
+  for (let lineIndex = 0; lineIndex < doc.length; lineIndex++) {
+    if (inValidationBlankNodes) {
+      doc[lineIndex] = '\t' + doc[lineIndex];
+    }
+
+    if (
+      prefixRegex.test(doc[lineIndex]) &&
+      !prefixRegex.test(doc[lineIndex + 1])
+    ) {
+      doc[lineIndex] = doc[lineIndex] + '\n';
+      continue;
+    }
+
+    if (
+      predicateWithNodeValueRegex.test(doc[lineIndex]) ||
+      predicateWithStringOrNumberValueRegex.test(doc[lineIndex]) ||
+      aSubbjectRegex.test(doc[lineIndex])
+    ) {
+      doc[lineIndex] = '\t' + doc[lineIndex];
+
+      if (doc[lineIndex].endsWith('.')) {
+        doc[lineIndex] = doc[lineIndex] + '\n';
+      }
+      continue;
+    }
+
+    if (
+      startOfValidationsRegex.test(doc[lineIndex]) ||
+      (lineIndex !== 0 && doc[lineIndex].endsWith('['))
+    ) {
+      doc[lineIndex] = '\t' + doc[lineIndex];
+      inValidationBlankNodes = true;
+      continue;
+    }
+
+    if (endBlankNodeRegex.test(doc[lineIndex])) {
+      inValidationBlankNodes = false;
+      doc[lineIndex] = '\t' + doc[lineIndex];
+    }
+
+    if (lineIndex !== 0 && doc[lineIndex - 1].endsWith(',')) {
+      doc[lineIndex] = '\t' + doc[lineIndex];
+    }
+  }
+}
+
+const prefixRegex = new RegExp(/@prefix\s+([^:]+):\s+<([^>]+)>\s*\.*/);
+const unknownPrefixRegexPattern = new RegExp(/@prefix\s+:\s+<#>\./);
+const aSubbjectRegex = new RegExp(/a\s+(\w+):(\w+)[;,]?/);
+const subjectWithTypeRegex = new RegExp(
+  /(\w+):([\w-]+)\s+a\s+(\w+):(\w+)[;,]?/
+);
+const predicateWithNodeValueRegex = new RegExp(
+  /(\w+):(\w+)\s+(\w+):\s*([^;\n\s]+)\s*[,;\.]/
+);
+const predicateWithStringOrNumberValueRegex = new RegExp(
+  /(\w+):(\w+)\s+("([^"]+)"|(\d+)|(true|false)|"""([^"]+)""")\s*[;,\.]/
+);
+const startOfValidationsRegex = new RegExp(/(\w+):([\w-]+)\s[\[]/);
+const lineBeforeValidationSummation = new RegExp(/.*],/);
+const endBlankNodeRegex = new RegExp(/[\]]\s*;/);
```

Comment:
Lets see what type of emojis we get out of this!

Reply:
![image](https://github.com/lblod/frontend-form-builder/assets/36266973/8ebf8c3f-29ec-4c19-86b0-c552c67e3f56)

Reply:
somewhat disappointing

Reply:
expected them to be all over the place

Reply:
haha to bad 

You can maybe create new ones for these combinations ;p

Reply:
Should be fixed 
![image](https://github.com/lblod/frontend-form-builder/assets/36266973/3ad1e50f-2e8c-4f48-a7f8-066d54c88703)